### ISSUE TITLE
Hyperparameters should be a mutable struct (not a named tuple)

### ISSUE THREAD

**Issue by DilumAluthge** [cscherrer/SossMLJ.jl#119]
That way we can do hyperparameter optimization with MLJ.

**Comment by cscherrer** [comment]
Let's make it a `Dict{Symbol, Any}` or similar, then we can easily convert to a named tuple to use in the Soss model

**Comment by cscherrer** [comment]
Oh but does it have to be a mutable struct for MLJ? Maybe we just make a little wrapper? Would be annoying for users to have to do it from scratch each time

**Comment by DilumAluthge** [comment]
Yeah MLJ needs to be able to access with `getproperty` and `setproperty!`

What if we make a mutable struct that wraps a `Dict{Symbol, Any}`, and converts `getproperty` and `setproperty!` to the appropriate Dict methods.

**Comment by cscherrer** [comment]
Yep, I like it.

**Comment by DilumAluthge** [comment]
https://github.com/JuliaCollections/PropertyDicts.jl

**Comment by DilumAluthge** [comment]
Currently https://github.com/JuliaCollections/PropertyDicts.jl already supports getproperty.

Once that package is registered, we could make a PR to add support for setproperty! for Symbol keys.
